metals/src/main/scala/scala/meta/internal/metals/codeactions/ImportMissingSymbol.scala

```diff
@@ -46,7 +46,7 @@ class ImportMissingSymbol(compilers: Compilers) extends CodeAction {
     Future
       .sequence(params.getContext().getDiagnostics().asScala.collect {
         case d @ ScalacDiagnostic.SymbolNotFound(name)
-            if d.getRange().encloses(params.getRange().getEnd()) =>
+            if params.getRange().overlapsWith(d.getRange()) =>
```

Comment:
we cannot use `encloses` because sometimes the code action range is smaller than the diagnostic range, for example:

```scala
val x = Future.successful(2)
//      ^^^^^^  diagnostic range
//       <-->   code action range
```

or even partially overlapping

```scala
val x = Future.successful(2)
//      ^^^^^^  diagnostic range
//         <------->   code action range
```